### PR TITLE
feat: get w3c feature flags android

### PR DESCRIPTION
**Pull request by kholood-ea**
## Description of the change

Get W3C feature flags for android 

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

[JiraID:MOB14515
](https://instabug.atlassian.net/browse/MOB-14515)

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests

### Code review

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: android/src/test/java/com/instabug/reactlibrary/RNInstabugAPMModuleTest.java [hunk 1]
```diff
@@ -204,4 +206,29 @@ public void testSetFlowAttribute() {
         verify(APM.class, times(1));
         APM.endUITrace();
     }
+    @Test
+    public void givenCallback$isW3ExternalTraceIDEnabled_whenQuery_thenShouldCallNativeApiAndResolvePromise(){
```
Comment (AndrewAminInstabug) [inline android/src/test/java/com/instabug/reactlibrary/RNInstabugAPMModuleTest.java:210]: Greate effort, Here's a small request. Can we use the new naming convetions of unit test methods?

File: android/src/test/java/com/instabug/reactlibrary/RNInstabugAPMModuleTest.java [hunk 2]
```diff
@@ -204,4 +206,29 @@ public void testSetFlowAttribute() {
...
+        Promise promise = mock(Promise.class);
+        InternalAPM internalAPM = mock(InternalAPM.class);
+        apmModule.isW3ExternalTraceIDEnabled(promise);
+        boolean expected=internalAPM._isFeatureEnabledCP(APMFeature.W3C_EXTERNAL_TRACE_ID,"");
+       verify(promise).resolve(expected);
+    }
+    @Test
+    public void givenCallback$isW3ExternalGeneratedHeaderEnabled_whenQuery_thenShouldCallNativeApiAndResolvePromise(){
```
Comment (AndrewAminInstabug) [inline android/src/test/java/com/instabug/reactlibrary/RNInstabugAPMModuleTest.java:218]: same here

File: android/src/test/java/com/instabug/reactlibrary/RNInstabugAPMModuleTest.java [hunk 3]
```diff
@@ -204,4 +206,29 @@ public void testSetFlowAttribute() {
...
+        Promise promise = mock(Promise.class);
+        InternalAPM internalAPM = mock(InternalAPM.class);
+        apmModule.isW3ExternalGeneratedHeaderEnabled(promise);
+        boolean expected=internalAPM._isFeatureEnabledCP(APMFeature.W3C_GENERATED_HEADER_ATTACHING,"");
+        verify(promise).resolve(expected);
+    }
+    @Test
+    public void givenCallback$isW3CaughtHeaderEnabled_whenQuery_thenShouldCallNativeApiAndResolvePromise(){
```
Comment (AndrewAminInstabug) [inline android/src/test/java/com/instabug/reactlibrary/RNInstabugAPMModuleTest.java:226]: and here as well

**Comment by AndrewAminInstabug** [comment]
Please resolve the Circle CI failed jobs in order to be able to merge this PR.

**Review by AndrewAminInstabug — APPROVED** [review]

**Comment by kholood-ea** [comment]
@AndrewAminInstabug CI jobs is failing due to integrating with android snapshot, and will be fixed when w3c changes is released

**Pull request merged**